crates/cli/src/subcommands/publish.rs

```diff
@@ -198,7 +199,8 @@ pub async fn exec(config: Config, args: &ArgMatches) -> Result<(), anyhow::Error
             ));
         }
         PublishResult::PermissionDenied { domain } => {
-            let identity = decode_identity(&config)?;
+            let token = get_login_token_or_log_in(&mut config, server, !force).await?;
```

Comment:
this has a weird side effect: if we're not logged in, and we use `--anon-identity`, then a permission denied error will prompt the user to log in :sweat_smile:

Reply:
this is an exacerbation of an issue we've had for a while (namely a permission denied error _used_ to just say "error: you're not logged in. please log in"). I feel like I should add an `if anon_identity` case here too?